### PR TITLE
Update Build System

### PR DESCRIPTION
**Pull request by darthtrevino**
* Ignore `src/` in **.npmignore**; `dist/` contains the source the client is expected to use
* Ignore `dist/` in **.gitignore**, since it will be built on `npm test`
* Add an `npm test` script to drive Jest and Mocha
* Add a `yarn.lock` file

### CODE REVIEW

**Comment by darthtrevino** [comment]
@mthuret After this PR, I'll make another PR to add a `lint` task using **eslint**. Do you have a preferred rule set? I usually go with the AirBnb rules.

File: package.json [hunk 1]
```diff
@@ -39,13 +41,17 @@
...
     "react-addons-test-utils": "^0.14.7 || ^15.0.0",
     "react-dom": "^15.3.0",
-    "shelljs": "^0.7.3"
+    "react-test-renderer": "^15.3.1",
+    "rimraf": "^2.5.4"
   },
   "dependencies": {
-    "aphrodite": "^0.5.0"
+    "aphrodite": "^0.5.0",
+    "expose-members-loader": "0.0.3"
```
Comment (mthuret) [inline package.json]: I guess `expose-members-loader` and `rimraf` are useful for remaining devs ?

**Comment by mthuret** [comment]
Airbnb lint rules are fine with me.

**Comment by darthtrevino** [comment]
@mthuret `rimraf` is there so the `npm clean` task can work cross platform in Windows. The `expose-members-loader` was there by accident in this PR.

**Comment by mthuret** [comment]
If you try to run the jest test, it will not work. I'll fix that

**Pull request merged**